client/src/RokuDevice.ts

```diff
@@ -102,26 +107,40 @@ export class RokuDevice {
 		return updatedContents;
 	}
 
-	public sendEcpPost(path: string, params = {}, body: needle.BodyData = ''): Promise<needle.NeedleResponse> {
-		return this.sendEcp(path, params, body);
+	public sendEcpPost(path: string, params = {}, body: needle.BodyData = '', options: HttpRequestOptions = {}): Promise<needle.NeedleResponse> {
```

Comment:
Could we change the third parameter to be a single options object instead of needing to add a 4th param? I see you set `needle.bodyData = ''`. Will it always be a string? If so, we could maintain backwards compatibility by doing something like this:

```typescript
public sendEcpPost(path: string, params = {}, options: SomethingHere) {
    let body: needle.BodyData;
    let actualOptions: HttpRequestOptions;
    if (typeof options === '') {
        body = options;
     } else {
         body = options?.body ?? '';
         actualOptions = options;
    }
```

Reply:
No body can be any of `Buffer | KeyValue | NodeJS.ReadableStream | string | null;` Often times it is passed as an object and it will automatically add the json header and encode it for you.

Reply:
okay. well then, I guess it's probably gotta stay the way you implemented it.